### ISSUE TITLE
surrogate: add hold-out evaluation and cross-validated metrics (not train-set RMSE only)

### ISSUE THREAD

**Issue by rosspeili** [ARPAQLS/spinq-vqe#20]
## The gap

After #5, the surrogate workflow is:

1. Load 12 materials from CSV  
2. Train MLP on **all** of them  
3. Predict on the **same** 12  
4. Report training RMSE (~0.006) and plot predicted vs actual  

That plot will always look excellent with so few points. It does **not** demonstrate predictive power — only memorization. QAOA then uses `theta_oracle = theta_pred` from this in-sample fit, so optimizer comparisons are built on an oracle whose generalization is unproven.

## What to build

Introduce proper ML evaluation in `surrogate.py` and NB04:

**A. Cross-validation on the training corpus**

- Use `sklearn.model_selection.cross_val_score` (already optional-imported) with **leave-one-out** or **k-fold** (k≤5 for small N).
- Surface `cv_r2` / `cv_rmse` in `TrainedSurrogate` and `surrogate_summary()` — today CV R² often shows `n/a`.

**B. Explicit train / hold-out split**

- Add something like `train_surrogate(..., hold_out_frac=0.2)` or a fixed hold-out index list.
- Train on subset; predict on held-out materials only for the “generalization” figure.
- Keep training-set diagnostics separate from hold-out metrics in notebook output.

**C. Oracle policy for QAOA**

Document and optionally implement two modes:

| Mode | Use case |
|------|----------|
| `oracle='in_sample'` | Current behaviour — fine for pipeline debugging |
| `oracle='cv_predict'` | Each material’s θ_SH from out-of-fold predictions — harder, more honest |

For small N, LOOCV predictions are the most defensible “honest oracle” without waiting for issue #6 (larger dataset).

### Example API sketch

```python
sr = surrogate.train_surrogate(ds, cv_folds=5, random_state=0)
surrogate.surrogate_summary(sr)  # reports cv_rmse, cv_r2

theta_oracle = surrogate.predict_oracle(sr, ds, mode="loocv")
```

### End state

- NB04 prints both **train RMSE** and **CV / hold-out RMSE**
- `surrogate_predictions.png` can show train vs hold-out (different markers) or a dedicated `surrogate_cv.png`
- README / OVERVIEW can cite CV metrics instead of implying the MLP “predicts θ_SH” from structure alone

### Defensibility

This directly answers: *“How do you know the surrogate isn’t overfitting?”* Without it, the QAOA comparison (greedy beats QAOA) is still valid as a **workflow** result but weak as a **materials informatics** result. Hold-out evaluation separates “we built a QAOA demo” from “we built a screening method.”

### Tests

- `test_train_surrogate` asserts `cv_r2` is finite when sklearn available and N≥5
- Mock dataset with known linear relationship → CV metric sanity check

### Depends on

Larger N (issue on dataset expansion) makes CV much more meaningful; can still ship LOOCV on 12 rows as a first step.
